angular-ngrx-scss/src/app/user/services/user.service.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map, Observable } from 'rxjs';
+import { TokenService } from 'src/app/auth/services/token.service';
+import { UserApiResponse, UserState } from 'src/app/state/user';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  userToken = this.tokenService.getToken();
```

Comment:
This only gets fetched on initial loading of `UserService`, so if a new token is written to local storage then it will not be reflected here. I would recommend querying for the token in an HTTP interceptor that adds in the auth header.

I believe this is the root cause of the unauthorized errors you're getting when the home page loads for the first time post-redirect.